Lib/stat.py

```diff
@@ -156,13 +158,17 @@ def S_ISWHT(mode):
 def filemode(mode):
     """Convert a file's mode to a string of the form '-rwxrwxrwx'."""
     perm = []
-    for table in _filemode_table:
+    for index, table in enumerate(_filemode_table):
         for bit, char in table:
             if mode & bit == bit:
                 perm.append(char)
                 break
         else:
-            perm.append("-")
+            if index == 0:
```

Comment:
```suggestion
            if not perm:
```
And you can remove `index`.

Reply:
Oh right, feel free to write a follow-up PR if you want, sadly you reviewed the change just after it got merged automatically. I have not preference between ``enumerate()`` + ``if index == 0`` and ``not perm``.

Reply:
``_stat.filemode()`` and ``stat.filemode()`` implementations are very different. I tried to write a minimum fix.

Reply:
Ah, so there is a `_stat.filemode()`? I noticed that the Python implementation always use constants defined in Python, even if they are overwritten by `_stat` later. I was not sure whether this is good or bad. On one hand, the original use case was for tar files, which can came from other systems. On other hand, it should work with local files.